homework/index.js

```diff
@@ -30,17 +33,197 @@
+      class: 'card-group d-flex container-fluid',
+    });
+    createContributorCard(contributorsData, cardsGroup);
+  }
+
+  function createRepositoryWidget(container, options = {}) {
+    const widgetContainer = createAndAppend('div', container, {
+      class:
+        'd-flex flex-column  alert alert-primary justify-content-start align-items-center flex-sm-column flex-lg-row',
+    });
+    createAndAppend('i', widgetContainer, {
+      class: `widget-icon ${options.faClasses} fa-3x px-3 py-1`,
+    });
+    createAndAppend('h4', widgetContainer, {
+      text: `${options.title}:`,
+      class: 'py-2 px-2 widget-title',
+    });
+    createAndAppend(options.valueTag || 'h4', widgetContainer, {
+      text: options.value,
+      class: 'widget-value lead',
+    });
+    return widgetContainer;
+  }
+
+  function createRepositoryLayout(repositoriesData) {
+    const mainParent = document.getElementById('main');
+    const leftColumn = createAndAppend('div', mainParent, {
+      class: 'd-flex flex-column repo-detail',
+      id: 'repo-detail',
+    });
+
+    const widgetContainer = createRepositoryWidget(leftColumn, {
+      faClasses: 'fab fa-github',
+      title: 'Repository Name:',
+      value: repositoriesData.name,
+      valueTag: 'a',
+    });
+
+    const anchorTag = widgetContainer.lastChild;
+    anchorTag.setAttribute('href', repositoriesData.html_url);
+    anchorTag.setAttribute('target', '_blank');
+
+    createRepositoryWidget(leftColumn, {
+      faClasses: 'fas fa-pen-alt',
+      title: 'Description',
+      value: repositoriesData.description,
+    });
+
+    const createdDate = new Date(repositoriesData.created_at).toDateString();
+    createRepositoryWidget(leftColumn, {
+      faClasses: 'fas fa-calendar-plus',
+      title: 'Created',
+      value: createdDate,
+    });
+
+    const updatedDate = new Date(repositoriesData.updated_at).toDateString();
+    createRepositoryWidget(leftColumn, {
+      faClasses: 'fas fa-clock',
+      title: 'Updated',
+      value: updatedDate,
+    });
+
+    createRepositoryWidget(leftColumn, {
+      faClasses: 'fas fa-code-branch',
+      title: 'Forks',
+      value: repositoriesData.forks,
+    });
+  }
+
+  function appendRepositoriesToSelect(repositoriesData, cb) {
+    const selectList = document.getElementById('repo-select');
+    repositoriesData
+      .sort((a, b) => a.name.localeCompare(b.name))
+      .forEach((repository, index) => {
+        createAndAppend('option', selectList, {
+          text: repository.name,
+          value: index,
+        });
+      });
+
+    // dummy values
+    cb(repositoriesData[0], repositoriesData[0].contributors_url);
+
+    // create an eventlistener for select list options
+    selectList.addEventListener('change', () => {
+      selectList.parentElement.nextElementSibling.innerHTML = '';
+      cb(repositoriesData[selectList.value], repositoriesData[selectList.value].contributors_url);
```

Comment:
As per my comment in slack, you don't need this second parameter for your callback as the `contributors_url` is a property of the selected repository object that you pass as the first parameter. So this code could be simplified as:

```js
function appendRepositoriesToSelect(repositoriesData, cb) {
  ...
  cb(repositoriesData[0]);

  // create an eventlistener for select list options
  selectList.addEventListener('change', () => {
    selectList.parentElement.nextElementSibling.innerHTML = '';
    cb(repositoriesData[selectList.value]);
  });
}
```

See also line 174 below.